### PR TITLE
Dev to stage [IMPORTANT]

### PR DESCRIPTION
**Pull request by MortenMunk**
## Description

👏 Finally, I am able to present to you, a dev branch that should resemble staging, and a few added features such as linting, toast notifications, updated endpoints, refactors etc.

**THIS IS VERY IMPORTANT THAT YOU ALL CHECK YOUR FILES, AND THAT THE CODE IS HOW YOU WANT IT TO BE, AND THAT THE APP BEHAVES AS INTENDED. THIS PR IS CRUCIAL TO ELIMINATE CONFLICTS BETWEEN DEV AND STAGING.**

## Changes

- made urls in userApi.js and api.js identical to staging
- linting added to config files
- linting added to tests
- workflow updated in yml files
- eslintr updated to single quotes and semicolons
- singleIcon.svg removed
- singleIcon.png moved to images¨
- removed comments and stack.screen error screen in App.js **(Please check this is correct)**
- changed routes and return data in api.js  **(Please check this is correct)**
- changed routes and return data in userApi.js  **(Please check this is correct)**
- build changes added in app.js  **(Please check this is correct)**
- moved mp3 files to sound folder
- removed unused testvideo.mp4
- changed imports of video in content.js
- replaced hardcoded colors, as done in staging progressbar.js and customrating.js
- time added to explorecard.js
- added alert notification to base screen
- added dialogNotification.js
- added alert notification to educadoModal.js
- refactored import of icon in IconHeader.js
- Added toastNotification.js
- added format hours to coursecard.jsx
- dialogNotifications added to enterNewPasswordScreen
- completedCourses set in asyncstorage on LoginForm
- DialogNotification and alertNotification added to RegisterForm
- updated user info setting in registerForm
- Toast notification added to ResetPassword
- updated fetching on changeEmailButton, changeFirstNameButton, and changeLastNameButton
- refactor of deleteAccountButton
- Replaced icon with SVG in returnButton
- updated navigation in sectionCard
- welcomeSlider replaced icon with svg
- updated jestSetup to remove duplicate mocks from test files
- Changes in package.json  **(Please check this is correct)**
- toast notification added to coursescreen
- update on exerciseScreen and replaced hardcoded colors with tailwind colors
- removed keys from TextImageLecturerScreen  **(Please check this is correct)**
- updated fetching on Profile
- Welcome.js changed navigation **(Please check this is correct)**
- exercise completion functions added to UtilityFunctions
- updated endpoints in api.test.js
- updated snaps
- added jest.useRealTimers to mock setTimeout on LoginForm
- updated fields in exerciseScreenMock
- added mocks to CourseScreen test
- SectionScreen mocks added
- FormatHours function added to UtilityFunctions.test.js
- added tailwindcss to package.json, to fix error as pr. stackoverflow

## Related Issues

None.

## Checklist

- [X] Code has been tested locally and passes all relevant tests.
- [X] Documentation has been updated to reflect the changes, if applicable.
- [X] Code follows the established coding style and guidelines of the project.
- [X] All new and existing tests related to the changes have passed.
- [X] Any necessary dependencies or new packages have been properly documented.
- [X] Pull request title and description are clear and descriptive.
- [X] Reviewers have been assigned to the pull request.
- [X] Any potential security implications have been considered and addressed.
- [X] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

None.

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Branch: dev

Commit id: 1fd784b2965a2ea0fef1b0a270190e98f01c9a85

## Notes for Reviewers

**IT IS VERY IMPORTANT YOU CHECK YOUR OWN FILES, AND SEE THAT THEY ARE HOW THEY SHOULD BE. PLEASE ALSO CHECK YOUR FUNCTIONALITY BEHAVES AS INTENDED**

**SOME OF THE FUNCTIONALITIES FOR EXERCISE COMPLETION IS NOT YET ADAPTED TO THE NEW USER MODEL, BUT THIS WILL BE DONE THIS WEEK**

**SOME IMPORTS HAS BEEN CHANGED TO REQUIRE DUE TO LINTER ISSUES. THIS CAN BE FIXED AT A LATER POINT**

### CODE REVIEW

**Comment by PrometheusBork** [comment]
I have encountered 2 errors: 
1. When I have subscribed to a course and go to explore page it will show correctly that I am subscribed to fx "Financas Pessoais" but if I set filter to costura it will say i am subscribed to "Virtual Tutor Test"
2. The video doesn't work for me idk if that is a big problem right now, there is also a problem if you swipe too fast in lecturescreen, as you will basically break it.

File: eml/components/exercise/LeaveButton.js
Comment (sofiiagran) [inline eml/components/exercise/LeaveButton.js:1]: it replaces tailwind with hardcoded colors

File: eml/components/explore/CustomRating.jsx
Comment (sofiiagran) [inline eml/components/explore/CustomRating.jsx:1]: again replace tailwind config with hardcoded colors

**Review by PrometheusBork — APPROVED** [review]
It works fine for me, there are minor bugs though but it will be fixed in future pull requests. I think it is more important to get stage stable. Tested this with Staging backend btw

File: eml/components/login/RegisterForm.js [hunk 1]
```diff
@@ -158,11 +163,7 @@ export default function RegisterForm() {
 
     try {
```
Comment (sofiiagran) [inline eml/components/login/RegisterForm.js:164]: try {
      await registerUser(obj)
        .then(async function (response) {
          // saves input information locally
          await saveUserInfoLocally(response._id, firstName, lastName, email);
          await AsyncStorage.setItem(USER_ID, response._id);
        }).then(async function () {
        .then(async function () {
          // logs in the user, if no errors occur, navigates to home screen and sets token
          await loginFromRegister(obj);
        }) 

keep this from staging!!! It was a hotfix from the user test

File: eml/components/profileSettings/changeEmailButton.js
Comment (sofiiagran) [inline eml/components/profileSettings/changeEmailButton.js:1]: why not call it from StorageService?

File: eml/components/profileSettings/changeFirstNameButton.js
Comment (sofiiagran) [inline eml/components/profileSettings/changeFirstNameButton.js:1]: Why not call from StorageService?

File: eml/components/profileSettings/changeLastNameButton.js
Comment (sofiiagran) [inline eml/components/profileSettings/changeLastNameButton.js:1]: Why not call from StorageService?

File: eml/package.json [hunk 2]
```diff
@@ -11,7 +10,7 @@
     "test-coverage": "jest --coverage",
     "test-watch": "jest --watch",
     "test:components": "jest -i .*/components/explore.test.js -t \"Component Tests\"",
-    "build": "eas build -p android --profile preview",
```
Comment (sofiiagran) [inline eml/package.json:14]: keep this from staging:     "build": "eas build -p android --profile preview",

**Review by sofiiagran — CHANGES_REQUESTED** [review]
See my comments :))

+ ask @BenjaminVilladsen to fix the unique key thing, I believe it was directly committed to staging and therefore does not exist in this PR ...

File: eml/services/StorageService.js [hunk 3]
```diff
@@ -10,17 +10,8 @@ const USER_ID = '@userId';
 const USER_INFO = '@userInfo';
 
 export const getUserInfo = async () => {
-
   const fetchedUserInfo = JSON.parse(await AsyncStorage.getItem(USER_INFO));
```
Comment (amaliedilling) [inline eml/services/StorageService.js:13]: this should still have some type of error handling, no?

File: eml/App.js
Comment (amaliedilling) [inline eml/App.js:1]: we (gamified) use the ErrorScreen, but it should not be in the WelcomeStack

**Review by amaliedilling — CHANGES_REQUESTED** [review]
made some comments, pls have a look :))

File: eml/components/explore/CustomRating.jsx
Comment (carlryskov) [reply eml/components/explore/CustomRating.jsx:1]: We fix the key issues with our next PR

File: eml/screens/profile/Profile.js [hunk 4]
```diff
@@ -7,23 +7,22 @@ import {
...
+const USER_INFO = '@userInfo';
 
 export default function ProfileComponent() {
   const [firstName, setFirstName] = useState('');
   const [lastName, setLastName] = useState('');
 
-
   const getProfile = async () => {
     try {
-      const fetchedProfile = await getUserInfo();
```
Comment (PrometheusBork) [inline eml/screens/profile/Profile.js:21]: Again why are we not using getuserinfo from storageservice

**Review by carlryskov — APPROVED** [review]
The Video streaming related files look good for now. "Key" warning will be fixed once this PR is merged, and we can merge our own latest PR with necessary changes.

File: eml/api/api.js [hunk 5]
```diff
@@ -9,8 +9,9 @@ const digitalOcean = 'http://207.154.213.68:8888';
```
Comment (Donhusum) [inline eml/api/api.js:9]: I have reimplemented the error handling that was removed. I also removed a console log, and added a comment.

File: eml/api/userApi.js [hunk 6]
```diff
@@ -35,11 +35,11 @@ export const registerUser = async (obj) => {
     email: ${obj.email ?? 'undefined'}`);
```
Comment (Donhusum) [inline eml/api/userApi.js:35]: Added uniform error handling to all functions in userApi

**Review by amaliedilling — APPROVED** [review]

**Review by sofiiagran — APPROVED** [review]
I fixed my fixes

**Review by ITPini — APPROVED** [review]

**Pull request merged**